### PR TITLE
Model for feature list

### LINKED ISSUE #173: Model for feature list

### PR DESCRIPTION
**Pull request by akbhatta**
### Description

Model for feature list
GetFeatureChannelList interface/model is made FGAC enabled for supporting UI

[Tests]
Added unit tests for the model

Signed-off-by: @akbhatta

### Issues Resolved
#173 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: notifications/src/main/kotlin/org/opensearch/commons/notifications/action/GetPluginFeaturesRequest.kt [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import org.opensearch.common.xcontent.XContentParserUtils
+import org.opensearch.commons.notifications.NotificationConstants.COMPACT_TAG
+import org.opensearch.commons.utils.logger
+import java.io.IOException
+
+/**
+ * Action Request for getting notification plugin features.
+ */
+class GetPluginFeaturesRequest : ActionRequest, ToXContentObject {
+    val compact: Boolean // Dummy request parameter for transport request
```
Comment (joshuali925) [inline notifications/src/main/kotlin/org/opensearch/commons/notifications/action/GetPluginFeaturesRequest.kt:47]: why is this needed?

File: notifications/src/test/kotlin/org/opensearch/commons/notifications/action/GetPluginFeaturesResponseTests.kt [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            listOf("feature1", "feature2", "feature3"),
+            mapOf(
+                Pair("FeatureKey1", "FeatureValue1"),
+                Pair("FeatureKey2", "FeatureValue2"),
+                Pair("FeatureKey3", "FeatureValue3")
+            )
+        )
+        val jsonString = """
+        {
+            "config_type_list":["feature1", "feature2", "feature3"],
```
Comment (joshuali925) [inline notifications/src/test/kotlin/org/opensearch/commons/notifications/action/GetPluginFeaturesResponseTests.kt:87]: maybe `config1` instead of `feature1`? "feature" is something else in Notifications

**Review by joshuali925 — APPROVED** [review]

File: notifications/src/main/kotlin/org/opensearch/commons/notifications/action/GetPluginFeaturesRequest.kt (hunk 1 above)
Comment (akbhatta) [reply notifications/src/main/kotlin/org/opensearch/commons/notifications/action/GetPluginFeaturesRequest.kt:47]: REST and transport serialization/de-serialization would not have any items so added this as dummy bool entry in transport communication

File: notifications/src/test/kotlin/org/opensearch/commons/notifications/action/GetPluginFeaturesResponseTests.kt (hunk 2 above)
Comment (akbhatta) [reply notifications/src/test/kotlin/org/opensearch/commons/notifications/action/GetPluginFeaturesResponseTests.kt:87]: makes sense. will update.

**Review by zhongnansu — APPROVED** [review]
